src/datasets.js

```diff
@@ -322,13 +324,25 @@ export default {
         const parseResponse = (response) => {
             const contentType = response.headers['content-type'];
             const wrappedItems = wrapArray(response);
-            if (!disableBodyParser) wrappedItems.items = parseBody(wrappedItems.items, contentType);
+            try {
+                if (!disableBodyParser) wrappedItems.items = parseBody(wrappedItems.items, contentType);
+            } catch (e) {
+                throw new RetryableError(e);
+            }
             return wrappedItems;
         };
 
-        return requestPromise(requestOpts)
-            .then(parseResponse)
-            .catch(catchNotFoundOrThrow);
+        const getData = async () => {
```

Comment:
Originally I wanted to write that those 3 lines `request().then().catch()` are faster to read than these 7 `let...; try {...} catch{...} return...` but then a realized that this will probably produce much better stack trace which is better

Reply:
When `requestPromise` throws, `response` will be `undefined` and `parseResponse` will throw that it cannot get property headers of undefined. @petrpatek 

We should:
```js
try {
    const response = await requestPromise(requestOpts);
    return parseResponse(response);
} catch (err) {
    catchNotFoundOrThrow(err);
}
```

Reply:
Yeah, you are right. catchNotFoundOrThrow() returns null in a case of 404 which should be returned.

Reply:
@petrpatek could you add test for this?

Reply:
@mtrunkat Yes I will.